### DISCUSSION TITLE
When run migrate:fresh half way break, settings table not migrated, then prompt settings error for every command

### DISCUSSION THREAD

**Discussion by shiroamada** [spatie/laravel-settings#23] [Q&A] [answered]
When run `php artisan migrate:fresh` half way break, settings table not migrated, then my laravel project keep prompting settings error for every command I run. Even `php artisan`.

![image](https://user-images.githubusercontent.com/2062946/105278155-dee02c00-5bdf-11eb-94b1-7d621059b15c.png)

I have manually create the table and rerun the command. `php artisan migrate:fresh` not suppose to run everything even the table not exist.

I even run particular table also having the screenshot error.

`php artisan migrate --path=/database/migrations/2021_01_05_232441_create_settings_table.php`

Or any command I can run fix it?

**Comment by shiroamada** [comment]
May I know the correct work flow ? let's say I have new settings, if I added the in the GeneralSettings.php new setting and also the migration file. When I push to server. git pull is pulling the GeneralSettings.php, then I run `php artisan migrate`, it will prompt error.

```
Tried loading settings 'App\Http\Settings\GeneralSettings', and following properties were missing: new_setting

  at vendor/spatie/laravel-settings/src/Exceptions/MissingSettings.php:13
      9▕     public static function create(string $settingsClass, array $missingProperties, string $operation): self
     10▕     {
     11▕         $missing = implode(', ', $missingProperties);
     12▕
  ➜  13▕         return new self("Tried {$operation} settings '{$settingsClass}', and following properties were missing: {$missing}");
     14▕     }
     15▕ }
     16▕
```

How to avoid this?

**Comment by rubenvanassche** [comment, marked as answer]
Hi @shiroamada,

This is a limitation from the package at the moment, we've got the same problem in our own project for which this package was developed. Since Laravel registers all the dependencies at the boot process of your application the settings classes are loaded too. Which means theses classes will try to fetch the settings from database, when that database is not yet migrated the settings class will crash because it cannot find the correct properties.

At the moment there are two solutions to this:
1) Use resolve(YourSettingsClass::class) in the places where Laravel automatically autoloads the settings
2) Add the fields manually to your database so it's always in the correct state

I'm looking into this and I'm trying to make these settings lazy loaded, which means they will be loaded when a property is fetched but haven't find a proper solution for this.

**Reply by shiroamada** [reply]
No idea how solution 1 will work.
Current my solution is no.2 it will be easy.
Thanks for your help and looking for your solution. Cheers~

**Comment by shruko** [comment]
Found any Solution?

**Comment by mrmmg** [comment]
i have same problem in CI/CD, how can i fix this issue?

**Comment by delboy1978uk** [comment]
I just added a try catch, if the table doesnt exist, we know we haven't migrated yet, so just ignore the exception. 
Once I did that, I got the artisan command back, and could successfully migrate
It's a bit of a naff solution, but this works:

```php
        try {
            if ($this->getAppSettings()->rabbit_enabled === false) {
                return true;
            }
        } catch (QueryException $e) {
            if (\strstr($e->getMessage(), 'Base table or view not found')) {
                // settings have not been migrated yet, simply return true
                return true;
            }
        }
```
For me, this code relates to a Filament page being registered, and was the only use case being called, so i know it was safe to ignore it this way
